asciidoc-confluence-publisher-maven-plugin/src/main/java/org/sahli/asciidoc/confluence/publisher/maven/plugin/AsciidocConfluencePublisherMojo.java

```diff
@@ -142,6 +161,42 @@ public void execute() throws MojoExecutionException {
         }
     }
 
+  private void applyUsernameAndPasswordFromSettings() throws MojoExecutionException {
+    if (this.serverId == null) {
+      throw new MojoExecutionException("'serverId' must be set, if username/password are not provided");
+    }
+
+    Server server = this.mavenSettings.getServer(this.serverId);
+    if (server == null) {
+      throw new MojoExecutionException(String.format("server with id '%s' not found in settings", serverId));
+    }
+
+    // take username from settings.xml if not provided
+    if (this.username == null) {
+      if (server.getUsername() == null) {
+        throw new MojoExecutionException(String.format("'username' neither defined by server '%s' nor provided", serverId));
+      } else {
+        this.username = server.getUsername();
+      }
+    }
+
+    // take password from settings.xml if not provided
+    try {
+      if (this.password == null) {
+        if (server.getPassword() == null) {
+          throw new MojoExecutionException(String.format("'password' neither defined by server '%s' nor provided", serverId));
+        } else {
+          if (securityDispatcher instanceof DefaultSecDispatcher) {
+            ((DefaultSecDispatcher) securityDispatcher).setConfigurationFile("~/.m2/settings-security.xml");
```

Comment:
Is this approach of specifying the path to `settings-security.xml` compatible non-linux based platforms, e.g. windows? Or would be reading the `user-home` system property better?

Reply:
ack, will change this